tests/integration_tests/style/test_markdown.py

```diff
@@ -32,3 +33,18 @@ def test_markdown_style():
     assert (
         not needs_format
     ), "Some markdown files need formatting. Either run `./tools/devtool sh mdformat .` in the repository root, or apply the above diffs manually."
+
+
+def test_markdown_internal_links():
+    """Make sure markdown internal links work"""
+
+    md_files = [Path(x) for x in utils_repo.git_repo_files(root="..", glob="*.md")]
+    for md_file in md_files:
+        txt = md_file.read_text(encoding="utf-8")
+        for link in re.findall(r"\[.+?\]\((?P<link>.+?)\)", txt, re.DOTALL):
```

Comment:
I don't think we use footnotes for internal links very often, but if it's easy to do, maybe we could check those too?

Reply:
hmm yeah we only have 3 at the moment.

```
% git grep -c "\[\^\d+\]" *.md
CONTRIBUTING.md:2
SPECIFICATION.md:4
```

Reply:
Yeah, maybe not worth the effort